### DISCUSSION TITLE
Some of the problems I have encountered in using doks

### DISCUSSION THREAD

**Discussion by KevinBlandy** [thuliteio/doks#401] [Q&A] [answered]
## 1. index.main.js load exception

![image](https://user-images.githubusercontent.com/17979625/126894619-80817da6-1f29-45a6-89e0-2be5fb527412.png)

## 2. Problems with displaying images in documents

The page where this problem occurs is [here](https://www.springcloud.io/docs/springcloud/spring-cloud-sleuth/#21-spring-cloud-sleuth-%E4%BB%8B%E7%BB%8D)

![image](https://user-images.githubusercontent.com/17979625/126894625-01ed0a94-6fd5-471f-8fa3-2a2e3a714637.png)

## 3. No syntax highlighting for code blocks in the document

I have declared the syntax in the code block, but it doesn't seem to work. And I found the warning log in the console.

![image](https://user-images.githubusercontent.com/17979625/126894645-267ce68e-59bb-4555-ba77-d48d0c5cfe42.png)

![image](https://user-images.githubusercontent.com/17979625/126894720-3b2b607e-ed6c-45ed-84e8-38fc4584a71e.png)

These issues do not affect the reading of the documentation at the moment, but I would be interested to know how to go about fixing it.

Thanks!

My documentation website is: `https://www.springcloud.io/`

**Comment by KevinBlandy** [comment]
Sorry, I forgot to clarify a situation. I had merged some PRs initiated by dependabot and they were all upgrade dependencies.

![image](https://user-images.githubusercontent.com/17979625/126894838-a60a4815-1550-442a-b4bc-83b98621eb65.png)

Is this the main cause of the problem?

Thanks!

**Comment by h-enk** [comment, marked as answer]
Hello @KevinBlandy, 

Ad 1) Could not reproduce it with a local production build of [KevinBlandy/springcloud.io](https://github.com/KevinBlandy/springcloud.io). Try running the build again.

Ad 2) You're missing the `img-fluid` class on images — see also: [Responsive images](https://getbootstrap.com/docs/5.0/content/images/#responsive-images). Simplest way to add the class is to use the [image markdown render hook](https://gohugo.io/getting-started/configuration-markup/#image-markdown-example). Add `./layouts/default/markup/render-image.html` w/ something like:

```html
<img class="img-fluid" src="{{ .Destination | safeURL }}" alt="{{ .Text }}" {{ with .Title}} title="{{ . }}"{{ end }} />
```

Note, Doks also comes w/ [image shortcodes](https://getdoks.org/docs/recipes/images/)

Ad 3) You forgot to [configure higlight.js](https://getdoks.org/docs/extensions/highlight-js/#3-configure). So, add e.g. java + xml:

```js
import java from 'highlight.js/lib/languages/java';

hljs.registerLanguage('java', java);
hljs.registerLanguage('xml', xml);
```

Note, you'll also need to add text, log + properties (and remove languages you don't need /use).

Note, you're also getting warnings on unescaped HTML in fenced code blocks. Solution: in `./config/_default/config.toml`, set `codeFences = false`.

**Reply by KevinBlandy** [reply]
Thank you very much for your patient answer.
